File: apps/desktop/src/types/customer.ts
export interface Customer {
  id: string;
  tenant_id: string;
  company_id: string;
  name: string;
  trade_name?: string | null;
  document?: string | null;
  phone?: string | null;
  email?: string | null;
  address_street?: string | null;
  address_number?: string | null;
  address_neighborhood?: string | null;
  city?: string | null;
  state?: string | null;
  postal_code?: string | null;
  notes?: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CustomerCreateInput {
  name: string;
  trade_name?: string;
  document?: string;
  phone?: string;
  email?: string;
  address_street?: string;
  address_number?: string;
  address_neighborhood?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  notes?: string;
  is_active?: boolean;
}

export interface CustomerUpdateInput extends Partial<CustomerCreateInput> {}

export interface Supplier {
  id: string;
  tenant_id: string;
  company_id: string;
  name: string;
  trade_name?: string | null;
  document?: string | null;
  state_registration?: string | null;
  phone?: string | null;
  email?: string | null;
  contact_person?: string | null;
  address_street?: string | null;
  address_number?: string | null;
  address_neighborhood?: string | null;
  city?: string | null;
  state?: string | null;
  postal_code?: string | null;
  notes?: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SupplierCreateInput {
  name: string;
  trade_name?: string;
  document?: string;
  state_registration?: string;
  phone?: string;
  email?: string;
  contact_person?: string;
  address_street?: string;
  address_number?: string;
  address_neighborhood?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  notes?: string;
  is_active?: boolean;
}

export interface SupplierUpdateInput extends Partial<SupplierCreateInput> {}

File: apps/api/app/models/customer.py
import uuid
from typing import TYPE_CHECKING
from sqlalchemy import String, Boolean, UUID, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.models.base import Base, TimestampMixin

if TYPE_CHECKING:
    from app.models.tenant import Tenant
    from app.models.company import Company


class Customer(Base, TimestampMixin):
    """
    Customer entity representing clients (PF/PJ) for sales and PDV.
    """
    __tablename__ = "customers"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4
    )
    tenant_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True
    )
    company_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("companies.id", ondelete="CASCADE"),
        nullable=False,
        index=True
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    trade_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    document: Mapped[str | None] = mapped_column(String(20), nullable=True, index=True) # CPF or CNPJ
    phone: Mapped[str | None] = mapped_column(String(30), nullable=True, index=True)
    email: Mapped[str | None] = mapped_column(String(255), nullable=True, index=True)
    
    # Address info
    address_street: Mapped[str | None] = mapped_column(String(255), nullable=True)
    address_number: Mapped[str | None] = mapped_column(String(20), nullable=True)
    address_neighborhood: Mapped[str | None] = mapped_column(String(100), nullable=True)
    city: Mapped[str | None] = mapped_column(String(100), nullable=True)
    state: Mapped[str | None] = mapped_column(String(2), nullable=True)
    postal_code: Mapped[str | None] = mapped_column(String(10), nullable=True)
    
    notes: Mapped[str | None] = mapped_column(String(512), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    # Relationships
    tenant: Mapped["Tenant"] = relationship("Tenant")
    company: Mapped["Company"] = relationship("Company")


class Supplier(Base, TimestampMixin):
    """
    Supplier entity representing vendors and distributors for purchase invoices.
    """
    __tablename__ = "suppliers"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4
    )
    tenant_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True
    )
    company_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("companies.id", ondelete="CASCADE"),
        nullable=False,
        index=True
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False, index=True) # Razao Social
    trade_name: Mapped[str | None] = mapped_column(String(255), nullable=True) # Nome Fantasia
    document: Mapped[str | None] = mapped_column(String(20), nullable=True, index=True) # CNPJ or CPF
    state_registration: Mapped[str | None] = mapped_column(String(30), nullable=True) # IE
    phone: Mapped[str | None] = mapped_column(String(30), nullable=True, index=True)
    email: Mapped[str | None] = mapped_column(String(255), nullable=True, index=True)
    contact_person: Mapped[str | None] = mapped_column(String(100), nullable=True)

    # Address info
    address_street: Mapped[str | None] = mapped_column(String(255), nullable=True)
    address_number: Mapped[str | None] = mapped_column(String(20), nullable=True)
    address_neighborhood: Mapped[str | None] = mapped_column(String(100), nullable=True)
    city: Mapped[str | None] = mapped_column(String(100), nullable=True)
    state: Mapped[str | None] = mapped_column(String(2), nullable=True)
    postal_code: Mapped[str | None] = mapped_column(String(10), nullable=True)

    notes: Mapped[str | None] = mapped_column(String(512), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    # Relationships
    tenant: Mapped["Tenant"] = relationship("Tenant")
    company: Mapped["Company"] = relationship("Company")
